lib/paranoia.rb

```diff
@@ -174,7 +174,7 @@ def really_destroy!
   private
 
   def each_counter_cached_associations
-    !@_disable_counter_cache && defined?(super) ? super : []
+    !(defined?(@_disable_counter_cache) && @_disable_counter_cache) && defined?(super) ? super : []
```

Comment:
Merci! I think you can also delete the `defined?(super)` check here as well if you want. We've dropped support for Rails 4.1 and this was meant for that old version.